src/Transition.js

```diff
@@ -44,51 +46,41 @@ class Transition extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.in && this.props.unmountOnExit) {
-      if (this.state.status === UNMOUNTED) {
-        // Start enter transition in componentDidUpdate.
-        this.setState({status: EXITED});
-      }
-    }
-    else {
-      this._needsUpdate = true
+    this._propsUpdated = true
+
+    if (nextProps.in && this.state.status === UNMOUNTED) {
+      // Start enter transition in componentDidUpdate.
+      this.setState({ status: EXITED });
     }
   }
 
   componentDidUpdate() {
     const status = this.state.status;
 
-    if (this.props.unmountOnExit && status === EXITED) {
-      // EXITED is always a transitional state to either ENTERING or UNMOUNTED
-      // when using unmountOnExit.
-      if (this.props.in) {
-        this.performEnter(this.props);
-      } else {
-        this.setState({status: UNMOUNTED});
-      }
+    // EXITED is always a transitional state to either ENTERING or UNMOUNTED
+    // when using unmountOnExit.
+    const needsTransitionalUpdate = (
```

Comment:
this is a little confusing because a transitional update is needed when `mountOnEnter` and `in` are both set